### ISSUE TITLE
How will the Simex Platform approach numerical and computational challenges in adding photons with phase information?

### ISSUE THREAD

**Issue by bussmann** [PaNOSC-ViNYL/SimEx#9]
When dealing with Monte Carlo photon sampling of scattering the number of photons and the correct addition of all photons according to their phases is a significant numerical challenge due to the precision required and a significant computational challenge due to the large number of photons required. How will the Simex Platform deal with these challenges?

**Comment by CFGrote** [comment]
Currently, all actual calculations are entirely left to the specialized calculators.

**Comment by bussmann** [comment]
Yes, but from the design point of view it will be necessary to decide where in the workflow chain the calculators take over. Just remember that in some way you will have to have the phase information of the photons that should be transported and scattered by the calculator. We find that this information is needed at a VERY HIGH numerical accuracy and that in order to correctly smaple the scattering one will need A LOT OF photons. One could of course create them from a (more or less sophisticated) parametrization of the XFEL beam inside the calculator, but I am not sure this is intended. I am discussing these issues now since they will later be hard or easy to solve depending on the choice of interface.

**Comment by CFGrote** [comment]
The photons are delivered by propagation calculators to the file or object that the PMI/scattering calculator reads from. The propagator will also query which data sets are required by the PMI calculator and the master calculator (PhotonExperimentSimulation) instance will only proceed with the workflow if these interface requirements are fulfilled.

**Comment by CFGrote** [comment]
It IS intended to provide calculators  of various degrees of sophistication and coarseness for each step in the pipeline. For the quick-and-dirty calculations a parameterized XFEL source/propagation calculator can be used, and for the more fine grained studies, the full numerical propagation.
